### PR TITLE
Add Layers

### LINKED ISSUE #8: Layer stack

### PR DESCRIPTION
**Pull request by RogueMacro**

### CODE REVIEW

**Review by jamesaorson — COMMENTED** [review]
Can we have a quick explanation of what this is for?

**Review by jamesaorson — CHANGES_REQUESTED** [review]
Shouldn't a LayerStaxk provide a proper pop and push? Since it is for ordering events, just allowing people to add and remove from arbitrary locations doesn't provide anything valuable.

**Comment by RogueMacro** [comment]
I will adress this. Quick explanation: there will be a layer for ie the game world, one for ui and so on. When an event occurs it gets passed down the layer stack and layers can consume events.

File: Engine/src/LayerStack.bf [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		public void PopLayer(Layer layer)
+		{
+			_layers.Remove(layer);
+			_layerInsert--;
+		}
+
+		public void PopOverlay(Layer overlay)
+		{
+			_layers.Remove(overlay);
+		}
```
Comment (jamesaorson) [inline Engine/src/LayerStack.bf:29]: Shouldn't these use `PopBack()` if they are gonna act as a Stack?

**Review by jamesaorson — CHANGES_REQUESTED** [review]

**Comment by RogueMacro** [comment]
Accidentally made another pull request, just move over there instead: https://github.com/Beef-Community-Project/Steel/pull/18

**Pull request closed without merging**